lib/rets/base/core.rb

```diff
@@ -254,7 +255,8 @@ def search(args, &block)
           raise RETS::CapabilityNotFound.new("Cannot find URL for Search call")
         end
 
-        req = {:url => @urls[:search], :read_timeout => args[:read_timeout], :open_timeout => args[:open_timeout]}
+        http_method = args[:http_method] || 'GET'
+        req = {:url => @urls[:search], :read_timeout => args[:read_timeout], :open_timeout => args[:open_timeout], method: args[:http_method]}
```

Comment:
You're setting the `method` option here to `args[:http_method]` instead of `http_method`.